### PR TITLE
ENH: Added example notebook for parameter space visualization with ExhaustiveOptimizerv4

### PR DESCRIPTION
**Pull request by tbirdso**

### CODE REVIEW

**Review by thewtex — CHANGES_REQUESTED** [review]
Very nice!

Please add a *requirements.txt* file with the required packages, and a [Binder](https://mybinder.org/) link in *Documentation.rst*.

**Comment by tbirdso** [comment]
@thewtex The `ExhaustiveOptimizerv4` example relies on wrapping included in `v5.2.rc02` but the latest Python `pip` distribution of ITK is `v5.1.2`. Is Binder able to build ITK or would this fail until the `pip` distribution is updated?

**Comment by thewtex** [comment]
@tbirdso `itk5.2rc2` is now available on PyPI. It can be installed locally with the `--pre` flag, e.g. `pip install --upgrade --pre itk`. We can add a version specification in the *requirements.txt* file for Binder, e.g. `itk>=5.2rc2`.

**Comment by tbirdso** [comment]
On testing this it seems that the `Binder` link points to files in a given commit or branch, which means the location would change if we add the `Binder` link in the same PR as the notebook it references. Should the `Binder` documentation reference be added in a separate PR once the example notebook is added to the `master` branch?

**Comment by tbirdso** [comment]
Rescinding my earlier comment, the Binder documentation link can point towards the ITKExamples `master` branch regardless of whether a given notebook has been merged in yet. I've tested the Binder link against the current `exhaustive-notebook` branch to verify functionality with this PR, then set the link to correctly point to `master` when the merge is completed.

Pending checks, this PR is ready for review and merge.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+pytest==6.2.1
+python-dateutil==2.8.1
+pyzmq==20.0.0
+six==1.15.0
+testpath==0.4.4
+toml==0.10.2
+tornado==6.1
+traitlets==5.0.5
+traittypes==0.2.1
+vtk==9.0.1
```
Comment (thewtex) [inline requirements.txt:63]: remove extra `vtk` dependency

File: src/Numerics/Optimizers/ExhaustiveOptimizer/Documentation.rst [hunk 2]
```diff
@@ -27,6 +27,15 @@ C++
 .. literalinclude:: Code.cxx
    :lines: 18-
 
+
+Jupyter Notebook
+-----------------
+
+.. image:: https://mybinder.org/badge_logo.svg
+ :target: https://mybinder.org/v2/gh/InsightSoftwareConsortium/ITKExamples/master
```
Comment (thewtex) [inline src/Numerics/Optimizers/ExhaustiveOptimizer/Documentation.rst:35]: This should point directly to the notebook:

https://mybinder.org/v2/gh/InsightSoftwareConsortium/ITKExamples/master?filepath=src%2FNumerics%2FOptimizers%2FExhaustiveOptimizer%2FPlotExhaustiveOptimizer.ipynb

**Review by thewtex — CHANGES_REQUESTED** [review]
Thanks for the updates.

A few issues identified inline.

File: src/Numerics/Optimizers/ExhaustiveOptimizer/Documentation.rst (hunk 2 above)
Comment (tbirdso) [reply src/Numerics/Optimizers/ExhaustiveOptimizer/Documentation.rst]: Fixed. Note that this link will not work until the merge completes and the notebook file exists in `master`. The equivalent link (while this branch exists only) is [https://mybinder.org/v2/gh/tbirdso/ITKExamples/exhaustive-notebook?filepath=src%2FNumerics%2FOptimizers%2FExhaustiveOptimizer%2FPlotExhaustiveOptimizer.ipynb](https://mybinder.org/v2/gh/tbirdso/ITKExamples/exhaustive-notebook?filepath=src%2FNumerics%2FOptimizers%2FExhaustiveOptimizer%2FPlotExhaustiveOptimizer.ipynb) , which I have verified launches and runs the notebook successfully.

**Comment by thewtex** [comment]
> Fixup: Remove VTK dependency and specify notebook Binder link 

Please squash the fixup commits into the previous commits with `git rebase -i`, i.e. Git interactive rebase. More information can be found here:

https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#revise-a-topic

**Comment by thewtex** [comment]
@tbirdso could this branch please be rebased on `master` to address the conflict?

**Comment by tbirdso** [comment]
@thewtex Updated with rebase onto `master`.

**Review by thewtex — APPROVED** [review]
:tada: :taco:

**Pull request merged**